packages/wallets/src/components/Base/WalletDialog/WalletDialog.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { isValidElement, PropsWithChildren } from 'react';
+import { Button, Dialog, Text } from '@deriv-com/ui';
+import './WalletDialog.scss';
+
+type TWalletDialogProps = {
+    cancelButtonText?: JSX.Element | string;
+    confirmButtonText?: JSX.Element | string;
+    hideCloseIcon?: boolean;
+    isVisible: boolean;
+    onCancel?: VoidFunction;
+    onClose: VoidFunction;
+    onConfirm?: VoidFunction;
+    shouldCloseOnOverlayClick?: boolean;
+    title?: JSX.Element | string;
+};
+
+const WalletDialog: React.FC<PropsWithChildren<TWalletDialogProps>> = ({
+    cancelButtonText,
+    children,
+    confirmButtonText,
+    hideCloseIcon = false,
+    isVisible = false,
+    onCancel,
+    onClose,
+    onConfirm,
+    shouldCloseOnOverlayClick = false,
+    title,
```

Comment:
Instead of passing the props here we can de-structure the components and pass the props based in the sub component instead.

```
<WalletDialog isVisible className='example-dialog' shouldCloseOnOverlayClick >
    <WalletDialog.Header className='example-dialog__header' hideCloseIcon>Title</WalletDialog.Header>
    <WalletDialog.Content className='example-dialog__content'>
          {/* Bla bla bla content */}
    </WalletDialog.Content>
   {isDesktop && (
       <WalletDialog.Footer>
          <Button onClick={() => {}}>Confirm</Button>
          <Button onClick={() => {}}>Cancel</Button>
       </WalletDialog.Footer>
    )}
</WalletDialog>
```